src/assertions/compiled.rs

```diff
@@ -212,46 +257,82 @@ enum Op {
         value: u16,
         operator: Comparison,
     },
-    // JsonPath {
-    //     value: JsonPathThingy,
-    // },
-    // XmlPath {
-    //     value: XmlPathThingy,
-    // },
+    JsonPath {
+        value: JpQuery,
+    },
     HeaderCheck {
         key: HeaderComparison,
         value: HeaderComparison,
     },
 }
 
+impl Eq for Op {}
+
 impl Op {
     fn eval(
         &self,
         status_code: u16,
         headers: &hyper::header::HeaderMap<HeaderValue>,
-        _body: &str,
-    ) -> bool {
-        match self {
-            Op::And { children } => children
-                .iter()
-                .all(|op| op.eval(status_code, headers, _body)),
-            Op::Or { children } => children
-                .iter()
-                .any(|op| op.eval(status_code, headers, _body)),
-            Op::Not { operand } => !operand.eval(status_code, headers, _body),
-            Op::StatusCodeCheck { value, operator } => match operator {
-                Comparison::LessThan => status_code < *value,
-                Comparison::GreaterThan => status_code > *value,
-                Comparison::Equal => *value == status_code,
-                Comparison::NotEqual => *value != status_code,
-            },
+        body: &[u8],
+        gas: &mut u32,
+    ) -> Result<bool, Error> {
+        let result = match self {
+            Op::And { children } => {
+                let mut result = true;
+                for child in children.iter() {
+                    result &= child.eval(status_code, headers, body, gas)?;
+                    if !result {
+                        break;
+                    }
+                }
+                result
+            }
```

Comment:
```suggestion
            Op::And { children } => children
              .iter()
              .map(|child| child.eval(status_code, headers, body, gas))
              .try_fold(true, |_, r| r)?
```

I think you can do something like this?